src/tls.c

```diff
@@ -91,12 +96,31 @@ enftun_tls_load_credentials(struct enftun_tls* tls,
     }
     enftun_log_debug("Loaded client TLS certificate %s\n", cert_file);
 
-    if (!(SSL_CTX_use_PrivateKey_file(tls->ctx, key_file, SSL_FILETYPE_PEM) ||
-          SSL_CTX_use_PrivateKey_file(tls->ctx, key_file, SSL_FILETYPE_ASN1)))
+    key_used =
```

Comment:
Adding `key_used` makes the control flow even more complicated here.

Before the TSS stuff, we had in pseudocode:
```
if (!(use_key_file(PEM) ||
     (use_key_file(ASN1))
{
    log(error);
    return error;
}
```
    
Let's keep that same simple control flow:
```
if (!(use_key_file(file, PEM) ||
     (use_key_file(file, ASN1) ||
     (tpm_use_key(file))
{
    log(error);
    return error;
}
```

Basically, move the check `enftun_tls_tpm_is_tpm_key()` (and the #ifdef discussed in my prior comment) inside of  `enftun_tls_tpm_use_key()`, before call `enftun_tls_tpm_engine_init()`.

Reply:
Done